### DISCUSSION TITLE
vue3在vite中配置为了避免组件冲突忽略自定义元素报错

### DISCUSSION THREAD

**Discussion by 766380494** [hellof2e/quark-design#263] [Q&A]
[vite] [plugin:vite:vue] At least one <template> or <script> is required in a single file component.

**Comment by xxxsf** [comment]
是否能否提供再多一点的信息？

**Reply by 766380494** [reply]
在vite.config.ts中引入 @vitejs/plugin-vue
然后plugins配置
	  vue({
      template: {
        compilerOptions: {
          isCustomElement: (tag) => tag.startsWith("quark-"),
        },
      },
    }),
就报上边那个错

**Comment by xxxsf** [comment]
<img width="1434" alt="image" src="https://github.com/hellof2e/quark-design/assets/14307551/f1922dce-a1b6-45b6-9287-a29c1155094d">

![image](https://github.com/hellof2e/quark-design/assets/14307551/671c0980-3858-44b6-bf8d-4d23abdf5795)

vite.config.js
```js
import { defineConfig } from 'vite'
import vue from '@vitejs/plugin-vue'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [vue({
    template: {
      compilerOptions: {
        isCustomElement: tag => tag.startsWith('quark-')
      }
    }
  })],
})
```

package.json
```js
"devDependencies": {
    "@vitejs/plugin-vue": "^4.1.0",
    "vite": "^4.3.2"
  }
```

我这里无法重现，要不你把代码脱敏上传，git 仓库地址发我再看下？

**Reply by 766380494** [reply]
https://gitee.com/afterglowr/test.git

uniapp项目

**Comment by xxxsf** [comment]
看了下，你这个报错与 `quarkd` 无关。你可以先尝试将组件库不要去加载，报错依旧存在，建议先排查这个工程本身问题。

ps：组件冲突忽略自定义元素报错只是一个控制台 ***警告***，不影响组件正常使用。
